cpp/src/parquet/encoding.cc

```diff
@@ -1486,7 +1486,7 @@ class DictDecoderImpl : public DecoderImpl, virtual public DictDecoder<Type> {
       return;
     }
     uint8_t bit_width = *data;
-    if (ARROW_PREDICT_FALSE(bit_width >= 64)) {
+    if (ARROW_PREDICT_FALSE(bit_width > 32)) {
       throw ParquetException("Invalid or corrupted bit_width");
```

Comment:
could we update the message here.  I guess since this is failing today, we probably could never read bit-widths greater then 32 but the spec under RLE encoding says:

> This length restriction was not part of the Parquet 2.5.0 and earlier specifications, but longer runs were not readable by the most common Parquet implementations so, in practice, were not safe for Parquet writers to emit.

Its not clear if this also applies to dictionary bit-width.  But I guess a dictionary of 2-Billion plus entries is probably untenable anways.

Reply:
I think this restriction is somewhat separate from the dictionary specific bitwidth restriction. The dictionary bit width restriction has been there since at least version 2.2 in 2013: https://github.com/apache/parquet-format/commit/ad2e4c438cdf080bf042a5330965e2eefb0caa65 . A bit width > 32 bits would also not be compatible with the num_values field in the header: https://github.com/apache/parquet-format/blob/54e53e5d7794d383529dd30746378f19a12afd58/src/main/thrift/parquet.thrift#L544

Also, parquet cpp uses int32_t internal for indices, so to support higher bitwidths would require a refactor (ex:   https://github.com/apache/arrow/blob/01855c791056b7f712e6df82acf97ad3ab7b823a/cpp/src/parquet/encoding.cc#L1582 )

Reply:
Updated message.